lib/decanter/parser.rb

```diff
@@ -16,17 +15,44 @@ def klass_or_sym_to_str(klass_or_sym)
         when Class
           klass_or_sym.name
         when Symbol
-          klass_or_sym.to_s.singularize.camelize
+          symbol_to_string(klass_or_sym)
         else
           raise ArgumentError.new("cannot lookup parser for #{klass_or_sym} with class #{klass_or_sym.class}")
         end.concat('Parser')
       end
 
       # convert from a string to a constant
       def parser_constantize(parser_str)
+        # safe_constantize returns nil if match not found
         parser_str.safe_constantize ||
-        "Decanter::Parser::".concat(parser_str).safe_constantize ||
-        raise(NameError.new("cannot find parser #{parser_str}"))
+          concat_str(parser_str).safe_constantize ||
+          raise(NameError.new("cannot find parser #{parser_str}"))
```

Comment:
I think Codeclimate is complaining about the syntax of the `raise` call here... Usually it's something like `raise NameError, "cannot find parser ..."`.